### PR TITLE
Implement some clang-tidy suggestions

### PR DESCRIPTION
**Pull request by lucafedeli88**
Inspired by @ax3l , I've used [clang-tidy](https://clang.llvm.org/extra/clang-tidy/) on WarpX enabling the following checks:

- 	[[modernize-use-equals-default]](https://clang.llvm.org/extra/clang-tidy/checks/modernize-use-equals-default.html)
- 	[[modernize-loop-convert]](https://clang.llvm.org/extra/clang-tidy/checks/modernize-loop-convert.html)
- 	[[bugprone-integer-division]](https://clang.llvm.org/extra/clang-tidy/checks/bugprone-integer-division.html)

In practice, I've used this command to compile:
`CXX=clang++ CC=clang ccmake ../. -DCMAKE_CXX_CLANG_TIDY="clang-tidy;-checks=-*,modernize-use-equals-default, modernize-loop-convert, bugprone-integer-division"`

This PR fixes all the issues found with `modernize-loop-convert`, some of the issues found with `modernize-use-equals-default` (there are some subtleties related to unique pointers that I need to understand before fixing them all in another PR), and one of the two issues found with `bugprone-integer-division`.

Concerning the other issue found with `bugprone-integer-division`, I left it as it was since I don't know if the integer division there
is intentional or not:
```
/home/luca/Projects/WarpX/Source/Filter/NCIGodfreyFilter.cpp:61:36: warning: result of integer division used in a floating point context; possible loss of precision [bugprone-integer-division]
    Real weight_right = m_cdtodz - index/tab_length;
```

### CODE REVIEW

File: Source/Utils/WarpXUtil.cpp [hunk 1]
```diff
@@ -288,7 +288,7 @@ int safeCastToInt(const amrex::Real x, const std::string& real_name) {
     std::string assert_msg;
     // (2.0*(numeric_limits<int>::max()/2+1)) converts numeric_limits<int>::max()+1 to a real ensuring accuracy to all digits
     // This accepts x = 2**31-1 but rejects 2**31.
-    if (x < (2.0*(std::numeric_limits<int>::max()/2+1))) {
+    if (x < (2.0*static_cast<double>(std::numeric_limits<int>::max()/2+1))) {
```
Comment (ax3l) [inline Source/Utils/WarpXUtil.cpp]: @dpgrote @NeilZaim I think you developed this.
Should this read
```suggestion
    using namespace amrex::literals;
    if (x < (2.0_rt*static_cast<amrex::Real>(std::numeric_limits<int>::max()/2+1))) {
```
?

Also, let's consider creating a temporary  `constexpr` variable before the `if` for the RHS of this inequality :)
```suggestion
    using namespace amrex::literals;
    constexpr amrex::Real max_range = (2.0_rt*static_cast<amrex::Real>(std::numeric_limits<int>::max()/2+1));
    if (x < max_range) {
```

**Comment by ax3l** [comment]
Regarding this line:
https://github.com/ECP-WarpX/WarpX/blob/e0085eb72f4f55d3bd3fb30af7490c999d34b9c7/Source/Filter/NCIGodfreyFilter.cpp#L61

I think @RemiLehe or @MaxThevenet might know that one.

Do we want to do the whole RHS as integer operations and then cast to float? Or only the second term? Or none at all and all as floating point?
We should add a `static_cast<amrex::Real>()` (or a simple `amrex::Real(...)`) at the appropriate location for this one :)

**Comment by NeilZaim** [comment]
All of these changes look good to me, thanks!

File: Source/Utils/WarpXUtil.cpp (hunk 1 above)
Comment (NeilZaim) [reply Source/Utils/WarpXUtil.cpp]: I think @dpgrote wrote this, but I personally like your suggestion (do everything as `amrex::Real` and use a `constexpr` variable).

**Review by ax3l — APPROVED** [review]
I'll approve this and pinged Rémi and Maxence again on your question for the other line you saw.

**Comment by RemiLehe** [comment]
@ax3l Regarding your question: my impression is that we want **none** of those operations to be done as integer operations (including the division in the 2nd term).

**Comment by MaxThevenet** [comment]
Good catch, I think all these operations should be on floats
```c++
Real weight_right = m_cdtodz - amrex::Real(index)/amrex::Real(tab_length); 
```
`m_cdtodz` is between 0 and 1, and we interpolate a set of coefficients from a table.

**Comment by MaxThevenet** [comment]
Sorry, I missed @RemiLehe's comment. I agree.

**Comment by ax3l** [comment]
Thanks a lot you two! :) 
I'll add this to the PR.

**Comment by ax3l** [comment]
Wuhu, the last one changes checksums - that means `clang-tidy` found an actual bug for `particles.use_fdtd_nci_corr = 1` :tada: 
- 2D: `PlasmaAccelerationBoost2d FAILED`
- 2D: `LaserAccelerationBoost FAILED`
- 3D: `PlasmaAccelerationBoost3d FAILED`
- 3D: `restart FAILED`

Will move to #2837, so we separate cleaning from bug fixes.

**Pull request merged**